components/camera/rtsp/h264decoder.go

```diff
@@ -0,0 +1,173 @@
+package rtsp
+
+// site:https://github.com/aler9/gortsplib/examples
+// author:Alessandro Ros
+
+// #cgo pkg-config: libavcodec libavutil libswscale
+// #include <libavcodec/avcodec.h>
+// #include <libavutil/imgutils.h>
+// #include <libswscale/swscale.h>
+import "C"
+
+import (
```

Comment:
Gotta still look through this but I'm curious why we are using this over mediadevices h264 codec

Reply:
I thought it was only encoding in mediadevices? Let me check again...

Reply:
from what I can tell, https://github.com/pion/mediadevices/blob/master/pkg/codec/x264/x264.go is just the encoder, but this is h264 -> image.RGBA decoding.

Reply:
ah. can we not use the x264 `c` code though so that we have just one dependency?

Reply:
Rather than changing the library to just the x264 one, what if I changed this PR to allow any codec available from libavcodec (which is from ffmpeg).It will make it more general for many different formats (Prob could have just done that from the beginning)
 
the available ones are 
 -H264
- HEVC (i.e. H265)
- MPEG4
- MPEG2video
- Theora
- VC-1 (i.e. SMPTE 421)
- RAW

Reply:
That might be okay but I'm afraid of how much bloat that will add to our packaging size. @Otterverse what do you think?

Reply:
It almost seems better as a modular resource

Reply:
So the footprint is pretty significant. Installing JUST those three libraries on a bare-bones debian system brings in 308MB of required other packages/files. Adding them to our AppImage doubles the _compressed_ size from 64MB to 128MB.

As to the two followup questions... I think supporting all the codecs would make a LOT of sense rather than just h264.

Lastly, yeah, due to the size... I think this would be a good candidate as a module. Or we acquire _enough_ new things capable of making use of libavcodec that it becomes worthwhile to put it in the main RDK.

OH, and yeah, hoping that module stuff is done any day now. I'm hoping to be ready for final reviews later today, but i've been saying that since thursday.

Reply:
Wait, but since these libraries are already included in ffmpeg, which we already have in RDK, would it still add to the overall size of RDK? I don't think this would require adding anything new to RDK. 

 Or does explictly using cgo (rather than just pointing to the ffmpeg binary like we do in the ffmpeg model) mean having to copy those files and repeat them?

Reply:
What if we removed the ffmpeg model, and just included these libraries. Would it decrease the size of RDK? Not that we can remove ffmpeg... but if people are only using ffmpeg for the RTSP capability, we maybe could...

Reply:
We don't include ffmpeg in the actual distribution/packaging, just libx264 and libopus. As far as I know, the ffmpeg camera model only calls an external binary. It never actually links to shared cgo code/libraries. But because we have that model, ffmpeg is needed for testing, and so is in the canon docker images.

The actual list of what gets bundled as true dependencies is here: https://github.com/viamrobotics/rdk/blob/main/etc/packaging/appimages/viam-server-x86_64.yml#L36-L41 (though those few packages pull in a couple more of their own dependencies.)